### PR TITLE
Issue #12421: tidy should also check .html files

### LINKED ISSUE #12421: ./mach test-tidy not checking .html files
Even though "./mach test-tidy" was successful for the patch in https://github.com/servo/servo/pull/12365, it was introducing several \t characters; as spotted in the code review.

That's because test-tidy does not consider .html files, while I believe it should. This is trivially fixed by adding *.html in FILE_PATTERNS_TO_CHECK, however this also requires to fix a bunch of existing files to keep "./mach test-tidy --all" clean.

I propose to do it via this issue.

### PR DESCRIPTION
**Pull request by simartin**
./mach test-tidy should also check .html files, to avoid patches containing badly formatted new test cases go unnoticed

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #12421
- [X] There are tests for these changes (./mach test-tidy failed without the changes to .hrml files)

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12576)

### CODE REVIEW

File: tests/html/tiny_test.html [hunk 1]
```diff
@@ -1 +1 @@
-
```
Comment (notriddle) [inline tests/html/tiny_test.html]: Just to be clear, this one's trailing space is not deliberate, right?

File: tests/wpt/mozilla/tests/css/pre_with_tab.html [hunk 2]
```diff
@@ -1 +1 @@
-<html><link rel='match' href='pre_with_tab_ref.html'><body><pre>	tab</pre></body></html>
```
Comment (notriddle) [inline tests/wpt/mozilla/tests/css/pre_with_tab.html]: This one should be a tab, not spaces.

**Comment by Wafflespeanut** [comment]
Why can't we run our WPT lint in those files?

File: tests/wpt/mozilla/tests/css/pre_with_tab.html (hunk 2 above)
Comment (simartin) [inline tests/wpt/mozilla/tests/css/pre_with_tab.html]: Yes indeed, thanks! Will correct in the next iteration of the patch

File: tests/html/tiny_test.html (hunk 1 above)
Comment (simartin) [inline tests/html/tiny_test.html]: I'm not sure; it was as such when the file was created via https://github.com/servo/servo/commit/c6ab60dbfc6da7b4f800c9e40893c8b58413960c. Will try to find out.

**Comment by simartin** [comment]
New patch version addressing all comments but the one about tests/html/tiny_test.html, that I still need to look at

File: tests/html/tiny_test.html (hunk 1 above)
Comment (jdm) [inline tests/html/tiny_test.html]: Correction, it was created in a3225428250a559c25af09f442e07e504d6e49b5 and the trailing space existed then, too. It's fine to remove it at this point.

**Comment by jdm** [comment]
@bors-servo: r+
Thanks @simartin!

**Comment by KiChjang** [comment]
Fails tidy:
```
Checking files for tidiness...

./tests/wpt/mozilla/tests/css/box_shadow_blur_fixed_ref.html:23: no newline at EOF

./tests/wpt/mozilla/tests/css/box_shadow_blur_fixed.html:24: no newline at EOF

./tests/html/test_target_pseudoselector.html:17: tab on line

./tests/html/test_target_pseudoselector.html:18: tab on line
```

**Comment by Wafflespeanut** [comment]
@simartin Can you fix the merge conflict? Next time, we'll ensure that this is prioritized for merging :)

**Comment by simartin** [comment]
I'll rebase tomorrow and update the PR

**Comment by Wafflespeanut** [comment]
Good to know :)

**Comment by simartin** [comment]
Rebased and new issues in tests/html/test_target_pseudoselector.html, tests/wpt/mozilla/tests/css/box_shadow_blur_fixed_ref.html and tests/wpt/mozilla/tests/css/box_shadow_blur_fixed.html fixed

**Comment by jdm** [comment]
@bors-servo: r+
Thanks!

**Comment by notriddle** [comment]
@bors-servo retry

* #12568

**Pull request merged**